Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -116,67 +127,457 @@ sample_triangles(const TriangleRange& triangles, double distance, OutputIterator
+                             get_const_property_map(vertex_point, tm)));
+}
 
+template< class Concurrency_tag,
+          class TriangleMesh,
+          class PointRange>
+double max_distance_to_triangle_mesh(const PointRange& points,
+                                     const TriangleMesh& tm)
+{
+   return max_distance_to_triangle_mesh<Concurrency_tag, TriangleMesh, PointRange>(points, tm, parameters::all_default());
+}
 
+/*!
+ *\ingroup PMP_distance_grp
+ *  Computes the approximated Hausdorff distance between `tm` and `points`
+ *
+ * @tparam PointRange a Range of `Point_3`
+ * @tparam TriangleMesh a model of the concept `FaceListGraph`
+ * @tparam NamedParameters a sequence of \ref namedparameters for `tm`
+ * @param tm the triangle mesh to compute the distance to
+ * @param points the point_set of interest.
+ * @param precision the precision of the approximated value you want.
+ * @param np a sequence of \ref namedparameters for `tm` among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `tm` \cgalParamEnd
+ *    \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `Kernel`\cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template< class TriangleMesh,
+          class PointRange,
+          class NamedParameters>
+double max_distance_to_point_set(const TriangleMesh& tm,
+                                 const PointRange& points,
+                                 const double precision,
+                                 const NamedParameters& np)
+{
+  typedef typename GetGeomTraits<TriangleMesh,
+                                 NamedParameters>::type Geom_traits;
 
-} } // end of namespace CGAL::Polygon_mesh_processing
+  typedef CGAL::Orthogonal_k_neighbor_search<CGAL::Search_traits_3<Geom_traits> > Knn;
+  typedef typename Knn::Tree Tree;
+  Tree tree(points.begin(), points.end());
+  CRefiner<Geom_traits> ref;
+  BOOST_FOREACH(typename boost::graph_traits<TriangleMesh>::face_descriptor f, faces(tm))
+  {
+    typename Geom_traits::Point_3 points[3];
+    typename boost::graph_traits<TriangleMesh>::halfedge_descriptor hd(halfedge(f,tm));
+    for(int i=0; i<3; ++i)
+    {        
+        points[i] = get(choose_param(get_param(np, vertex_point),
+                                     get_const_property_map(vertex_point, tm)),
+                        target(hd, tm));
+        hd = next(hd, tm);
+    }
+    ref.add(points[0], points[1], points[2], tree);
+  }
+  return ref.refine(precision, tree);
+}
+
+template< class TriangleMesh,
+          class PointRange>
+double max_distance_to_point_set(const TriangleMesh& tm,
+                                 const PointRange& points,
+                                 const double precision)
+{
+  return max_distance_to_point_set(tm, points, precision, parameters::all_default());
+}
+// convenience functions with default parameters
```

Comment:
move the comment above the previous function and pu the convenience function for `max_distance_to_triangle_mesh` here too.